openshift-hack/e2e/annotate/rules.go

```diff
@@ -85,7 +70,6 @@ var (
 
 			// TODO(node): configure the cri handler for the runtime class to make this work
 			"should run a Pod requesting a RuntimeClass with a configured handler",
-			"should reject a Pod requesting a RuntimeClass with conflicting node selector",
```

Comment:
I see that some of the tests from this commit are still present in the codebase.
- Why were they not matched?
- For the ones that are present, are we confident that they can be enabled? I suppose we should run payload tests to be sure.

Reply:
> For the ones that are present, are we confident that they can be enabled?

No tests are enabled by this PR; you can see that `openshift-hack/e2e/annotate/generated/zz_generated.annotations.go` itself is unmodified. The PR only removes rules that had no effect on the output.

> Why were they not matched?

In this case, it appears that the test's full name ends up being `"[sig-node] RuntimeClass should reject a Pod requesting a RuntimeClass with conflicting node selector"`, so it is already matched by the pattern `"RuntimeClass should reject"` above (old line 80 / new line 65).

The rule being removed here was added as part of the kube 1.16 rebase in https://github.com/openshift/origin/pull/23811, without noticing that it was redundant with the other rule which was added 4 months earlier.

The rule on line 65 was added on 2019-05-18 in https://github.com/openshift/origin/pull/22857 as a temporary hack until we had an updated RHCOS build:
```diff
+			// requires a 1.14 kubelet
+			"when the NodeLease feature is enabled",
+			"RuntimeClass should reject",
```
It was reverted one month later in https://github.com/openshift/origin/pull/23198 but then un-reverted two days later in https://github.com/openshift/origin/pull/23227. Then people forgot about it, until 2 years later Clayton did some cleanup and removed the comment and the _first_ line in https://github.com/openshift/origin/pull/24835, but by that point nobody remembered that the comment applied to the second line too, so it didn't get removed.

Removing `"RuntimeClass should reject"` _does_ change the generated output though, enabling 3 tests that are currently disabled.

But anyway, this sort of thing is not unique to the _redundant_ rules; we have all sorts of rules that were added "temporarily" 12 releases ago and then forgotten about forever. :slightly_frowning_face:

Reply:
Thanks for the thorough analysis

I suppose for the tests that have been flaking in the past, the current CI runs/analysis should be more relevant than the stale annotation skips. 

> No tests are enabled by this PR; you can see that openshift-hack/e2e/annotate/generated/zz_generated.annotations.go itself is unmodified. The PR only removes rules that had no effect on the output

right, payload runs won't help here...